src/userscores/view/exportscoremodel.cpp

```diff
@@ -0,0 +1,363 @@
+{
+    static const QHash<int, QByteArray> roles {
+        { RoleTitle, "title" },
+        { RoleIsSelected, "isSelected" },
+        { RoleIsMain, "isMain" }
+    };
+
+    return roles;
+}
+
+void ExportScoreModel::toggleSelection(int scoreIndex)
+{
+    if (!isNotationIndexValid(scoreIndex)) {
+        return;
+    }
+
+    QModelIndex modelIndex = index(scoreIndex);
+    m_selectionModel->QItemSelectionModel::select(modelIndex, QItemSelectionModel::Toggle);
+
+    emit dataChanged(modelIndex, modelIndex);
+}
+
+void ExportScoreModel::toggleAllSelections(bool select)
+{
+    QModelIndexList previousSelectedIndexes = m_selectionModel->selectedIndexes();
+    m_selectionModel->QItemSelectionModel::select(QItemSelection(index(0), index(
+                                                                     m_notations.size() - 1)),
+                                                  select ? QItemSelectionModel::Select : QItemSelectionModel::Deselect);
+    QModelIndexList newSelectedIndexes = m_selectionModel->selectedIndexes();
+
+    QSet<QModelIndex> indexesToUpdate(previousSelectedIndexes.begin(), previousSelectedIndexes.end());
+    indexesToUpdate = indexesToUpdate.unite(QSet<QModelIndex>(newSelectedIndexes.begin(), newSelectedIndexes.end()));
+
+    for (const QModelIndex& indexToUpdate : indexesToUpdate) {
+        emit dataChanged(indexToUpdate, indexToUpdate);
+    }
+}
+
+IMasterNotationPtr ExportScoreModel::masterNotation() const
+{
+    return context()->currentMasterNotation();
+}
+
+bool ExportScoreModel::isMainNotation(INotationPtr notation) const
+{
+    return notation == masterNotation()->notation();
+}
+
+bool ExportScoreModel::isNotationIndexValid(int index) const
+{
+    return index >= 0 && index < m_notations.size();
+}
+
+QList<int> ExportScoreModel::selectedRows() const
+{
+    QList<int> result;
+
+    for (const QModelIndex& index: m_selectionModel->selectedIndexes()) {
+        result << index.row();
+    }
+
+    return result;
+}
+
+int ExportScoreModel::getSelectionLength() const
+{
+    return selectedRows().size();
+}
+
+QString ExportScoreModel::getExportPath() const
+{
+    return m_exportPath.toQString();
+}
+
+void ExportScoreModel::setExportPath(QString path)
+{
+    m_exportPath = io::path(path);
+}
+
+io::path ExportScoreModel::getDefaultExportPath() const
+{
+    return configuration()->scoresPath() + "/" + masterNotation()->metaInfo().title + ".pdf";
+}
+
+QString ExportScoreModel::chooseExportPath()
+{
+    io::path requestedPath = interactive()->selectSavingFile(qtrc("userscores", "Export Score"), m_exportPath, getExportFliter());
+
+    if (!requestedPath.empty()) {
+        m_exportPath = requestedPath;
+    }
+
+    return m_exportPath.toQString();
+}
+
+void ExportScoreModel::setExportSuffix(QString suffix)
+{
+    m_exportPath = dirpath(m_exportPath) + "/" + basename(m_exportPath) + "." + suffix;
+}
+
+void ExportScoreModel::exportScores()
```

Comment:
I think it is better to move this function (and all related logic) to a separate class

Reply:
A class called ExportScoreController perhaps?

Reply:
Something like that, but better to replace the word "controller" with "service": ExportScoreService (or ExportNotationService). We try to use the word "controller" only in those classes that handle actions (see FileScoreController for example)